environs/bootstrap/bootstrap.go

```diff
@@ -355,9 +362,19 @@ func Bootstrap(ctx environs.BootstrapContext, environ environs.Environ, args Boo
 	// agent-version set anyway, to appease FinishInstanceConfig.
 	// In the latter case, setBootstrapTools will later set
 	// agent-version to the correct thing.
-	agentVersion := jujuversion.Current
 	if args.AgentVersion != nil {
 		agentVersion = *args.AgentVersion
+	} else {
+		// We will fall here when bootstrap command is given an
```

Comment:
we'll also get here when building/uploading tools. so either adjust the comment, or (possibly leading to clearer code?)
 - move this to just after the findPackagedTools; and
 - set agentVersion=forceVersion at the end of the block that does the build/upload logic

Reply:
Yes, I just wanted your eyes on this before I re-shuffle the code and added tests \o/
Thank you for the review :D